src/components/Pages/Assets/Tokens/LoadTokensData.tsx

```diff
@@ -0,0 +1,26 @@
+import { isEmpty } from 'lodash';
+import { memo, useEffect, useState } from 'react';
+
+import { TokenWithSupply } from '@helpers/interfaces';
+
+import { useAssets } from '@hooks/useAssets';
+
+import Tokens from './Tokens';
+
+const LoadTokensData = () => {
+  const [tokens, setTokens] = useState<TokenWithSupply[] | null>(null);
+  const { getAllTokensWithNativeAndSupply } = useAssets();
+
+  useEffect(() => {
+    getAllTokensWithNativeAndSupply().then(setTokens);
+  }, [getAllTokensWithNativeAndSupply]);
+
+  if (isEmpty(tokens)) {
+    return <>Gathering data... please wait</>;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  return <Tokens tokens={tokens!} />;
```

Comment:
I thought we came to an agreement on not to use "!", hence the rule even reminds of this.

Reply:
I would say this is an exemption to make our linter happy :)
A couple of lines above, I'm checking the tokens array is not empty `if (isEmpty(tokens)) {` so, it's not necessary to use `!` here, but somehow linter doesn't understand that

Reply:
ok, I found another way of shutting him up